tests/tvk-oneclick/tvk_oneclick_test.sh

```diff
@@ -0,0 +1,198 @@
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_helm(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Helm_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_namespace(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Namespace_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_operator(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Operator_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+cleanup() {
+  local rc=$?
+  
+  # cleanup namespaces and helm release
+  INSTALL_NAMESPACE=default
+  #shellcheck disable=SC2143
+  if [[ $(helm list -n "${INSTALL_NAMESPACE}" | grep "${INSTALL_NAMESPACE}") ]]; then
+    helm delete "${HELM_RELEASE_NAME}" --namespace "${INSTALL_NAMESPACE}"
+  fi
+
+  kubectl get validatingwebhookconfigurations -A | grep "${INSTALL_NAMESPACE}" | awk '{print $1}' | xargs -r kubectl delete validatingwebhookconfigurations || true
+  kubectl get mutatingwebhookconfigurations -A | grep "${INSTALL_NAMESPACE}" | awk '{print $1}' | xargs -r kubectl delete mutatingwebhookconfigurations || true
+
+  # NOTE: need sleep for resources to be garbage collected by api-controller
+  sleep 20
+
+  kubectl delete ns "${INSTALL_NAMESPACE}" --request-timeout 2m || true
+
+  kubectl get po,rs,deployment,pvc,svc,sts,cm,secret,sa,role,rolebinding,job,target,backup,backupplan,policy,restore,cronjob -n "${INSTALL_NAMESPACE}" || true
+
+  kubectl get validatingwebhookconfigurations,mutatingwebhookconfigurations -A | grep -E "${INSTALL_NAMESPACE}" || true
+
+  #Destroying virtual cluster created
+  # shellcheck disable=SC2154
+  vcluster delete "$build_id" -n default
+  helm delete "$build_id" --namespace default
```

Comment:
wrong sequence of deletion. delete vcluster at the end

Reply:
Corrected!